В этом блокноте приведено обучение нейросети. Снова запускать его не нужно. Веса обученной сети уже есть в папке. Для нового обучения необходимо иметь видеокарту с большим количеством видеопамяти, настроенную CUDA и достаточное количество времени

In [1]:
from ultralytics import YOLO
import os
from IPython.display import display, Image
from IPython import display
display.clear_output()
!yolo checks

Ultralytics YOLOv8.0.154  Python-3.11.4 torch-2.0.1+cu118 CUDA:0 (NVIDIA GeForce RTX 3070 Laptop GPU, 8192MiB)
Setup complete  (8 CPUs, 15.7 GB RAM, 366.2/471.6 GB disk)


In [ ]:
Для запуска переобучения необходимо указать актуальный путь до yaml файла с настройками.

In [2]:
!yolo task=detect mode=train model=yolov8n.pt data={'C:/Users/PyatnitskayaOT/PycharmProjects/yolov8/hakaton_05_24/data'}/data.yaml  epochs=50 imgsz=640 batch=4

Ultralytics YOLOv8.0.154  Python-3.11.4 torch-2.0.1+cu118 CUDA:0 (NVIDIA GeForce RTX 3070 Laptop GPU, 8192MiB)
engine\trainer: task=detect, mode=train, model=yolov8n.pt, data=C:/Users/PyatnitskayaOT/PycharmProjects/yolov8/hakaton_05_24/data/data.yaml, epochs=50, patience=50, batch=4, imgsz=640, save=True, save_period=-1, cache=False, device=None, workers=8, project=None, name=None, exist_ok=False, pretrained=True, optimizer=auto, verbose=True, seed=0, deterministic=True, single_cls=False, rect=False, cos_lr=False, close_mosaic=10, resume=False, amp=True, fraction=1.0, profile=False, freeze=None, overlap_mask=True, mask_ratio=4, dropout=0.0, val=True, split=val, save_json=False, save_hybrid=False, conf=None, iou=0.7, max_det=300, half=False, dnn=False, plots=True, source=None, show=False, save_txt=False, save_conf=False, save_crop=False, show_labels=True, show_conf=True, vid_stride=1, line_width=None, visualize=False, augment=False, agnostic_nms=False, classes=None, retina_masks=False, 

Тут продублированный результаты обучения

Ultralytics YOLOv8.0.154  Python-3.11.4 torch-2.0.1+cu118 CUDA:0 (NVIDIA GeForce RTX 3070 Laptop GPU, 8192MiB)
Model summary (fused): 168 layers, 3005843 parameters, 0 gradients

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95):   0%|          | 0/5 [00:00<?, ?it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95):  20%|##        | 1/5 [00:00<00:02,  1.90it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95):  40%|####      | 2/5 [00:01<00:01,  1.74it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95):  60%|######    | 3/5 [00:01<00:01,  1.92it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95):  80%|########  | 4/5 [00:02<00:00,  1.92it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|##########| 5/5 [00:02<00:00,  2.55it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|##########| 5/5 [00:02<00:00,  2.20it/s]
                   all         37       2954       0.98      0.951      0.983      0.884
Speed: 2.6ms preprocess, 7.7ms inference, 0.0ms loss, 1.5ms postprocess per image
Results saved to runs\detect\train3